api/src/main/java/jakarta/data/repository/Delete.java

```diff
@@ -55,12 +46,19 @@
  *     void unpark(Car car);
  * }
  * </pre>
- * <p>If this annotation is combined with other operation annotations (e.g.,
- * {@code @Find}, {@code @Insert}, {@code @Query}, {@code @Update},
- * {@code @Save}), it will throw an {@link UnsupportedOperationException} as only one operation type can be specified.</p>
- * <p>If the unique identifier of a requested entity is not found in the database or its version does not match, and the return
- * type of the annotated method is {@code void} or {@code Void}, the method must
- * raise {@link jakarta.data.exceptions.OptimisticLockingFailureException}.
+ * <p>Alternatively, the {@code Delete} annotation may be applied to a repository method with no parameters, indicating
+ * that the annotated method deletes all instances of the primary entity type. In this case, the annotated method must
+ * either be declared {@code void}, or return {@code int} or {@code long}.
+ * </p>
+ * <p>Deletes are performed by matching the unique identifier of the entity. If the entity is versioned, for example,
+ * with {@code jakarta.persistence.Version}, the version is also checked for consistency. Attributes other than the
+ * identifier and version do not need to match. If no entity with a matching identifier is found in the database, or
+ * if the entity with a matching identifier does not have a matching version, the annotated method must raise
+ * {@link jakarta.data.exceptions.OptimisticLockingFailureException}.
+ * </p>
+ * <p>If this annotation occurs alongside a different lifecycle annotation, the annotated repository method must raise
```

Comment:
@njr-11 I'm not a fan of this for the reason that we are begging the question of what makes two annotations "conflicting". We have not defined what that means. (And it's not enough to just give examples.)

IMO, the _specification_ and not the Javadoc of individual annotations is what really has the responsibility to say this. Otherwise we have to say it in lots of different places instead of in one place. So I almost removed these bits completely in favor of the spec addition here:

https://github.com/jakartaee/data/pull/490/commits/cac4ce1a147b96d2f9fed71617e2438fc6b433a7

So I think the issue is that you want to make clear that `@Find` can't be used together with `@Delete`. Well, if necessary the spec should say that. But when I thought about this quickly I thought that there were no method signatures which are legal for both `@Find` and `@Delete`. Perhaps that's over-optimistic.

Anyway the point is that if we need to say that `@Find` and query annotations conflict with lifecycle annotations, we should say it explicitly in the spec.

Reply:
(OK, I'm now proposing adding a new section to the spec to deal with this question.)

Reply:
Yes, go ahead and add the section to the spec.  Users typically won't read the spec, but will look at the JavaDoc of what they are using, so I do think it will help reduce user errors to call out what they cannot do.  I can definitely anticipate someone trying to annotate a method as both `@Query` and `@Delete` (or whatever type of query it is) because they think that makes sense to them.  I see what you are saying about not specifying in terms of examples.  I'll update my suggestions to fix that.  We can still call out not to combine with specific annotations.

Reply:
OK, if you definitely want it on the annotations as well, then fine.